File: modules/base/types/src/main/java/candle/types/resource/PreRemoveEvent.java
package candle.types.resource;

import candle.event.CancelableEvent;

public class PreRemoveEvent<T> extends CancelableEvent<Registry<T>> {
  private final Registry<T> registry;
  private final Identifier id;
  private final T entry;

  public PreRemoveEvent( Registry<T> registry, Identifier id, T entry ) {
    super(registry);
    this.registry = registry;
    this.id = id;
    this.entry = entry;
  }

  public Registry<T> getRegistry() {
    return registry;
  }

  public Identifier getId() {
    return id;
  }

  public T getEntry() {
    return entry;
  }
}

File: modules/base/types/src/main/kotlin/net/candlemc/types/resource/PreRemoveEvent.kt
package net.candlemc.types.resource

import net.candlemc.event.CancelableEvent

class PreRemoveEvent<T>(
    registry: Registry<T>, val id: Identifier,
    val entry: T
) :
    CancelableEvent<Registry<T>?>(registry) {
    private val registry: Registry<T> = registry

    fun getRegistry(): Registry<T> {
        return registry
    }
}
